### PR TITLE
Plotter compare

### PR DESCRIPTION
**Pull request by matthieuvigne**
This is a fixup of https://github.com/Wandercraft/jiminy/pull/93 : you forced push to my branch and thus removed the last commit.

There was code ducplication left in jiminy_py between `log.py` and `plotter.py` - but only `log.py` is meant to be used.

### CODE REVIEW

**Comment by duburcqa** [comment]
Arf ok sorry. I see in the diff that you are removing features, is this intended?

**Comment by duburcqa** [comment]
Perhaps it is time to create a dev branch ? I also have a minor patch to merge for windows (2 lines).

**Comment by matthieuvigne** [comment]
> Arf ok sorry. I see in the diff that you are removing features, is this intended?

No indeed, now I'm the one removing your changes :) I'll make a cleaner MR.

**Comment by matthieuvigne** [comment]
> Perhaps it is time to create a dev branch ? I also have a minor patch to merge for windows (2 lines).

I also have a minor patch for meshcat as pointed by Roch - so let's go with creating a dev branch for all of this :) I can have a look at it this afternoon.

File: python/jiminy_py/src/jiminy_py/log.py [hunk 1]
```diff
@@ -64,20 +68,17 @@ def read_log(filename):
...
-        "- If *h2* matches several fields : each field will be plotted individually in subplots. \n" + \
-        "- If :*h2* or :*h2*:*h3*:*h4* matches several fields : each field will be plotted in the same subplot. \n" + \
-        "- If *h2*:*h3*:*h4* matches several fields : each match of h2, h3, and h4 will be plotted jointly in subplots.\n" + \
-        "  Note that if the number of matches for h2, h3, h4 differs, only the minimum number will be plotted.\n" + \
-        "\nEnter no plot command (only the file name) to view the list of fields available inside the file."
-
-    parser = argparse.ArgumentParser(description=description_str,
-                                     formatter_class=argparse.RawTextHelpFormatter)
-    parser.add_argument("input", help="Input logfile.")
+    description_str = "Plot data from a jiminy log file using matplotlib.\n" + \
```
Comment (duburcqa) [inline python/jiminy_py/src/jiminy_py/log.py:71]: You should keep the "old" description no ?

File: python/jiminy_py/src/jiminy_py/log.py [hunk 2]
```diff
@@ -89,25 +90,27 @@ def plot_log():
...
+        for filename in main_arguments.compare.split(':'):
+            compare_data[filename], _ = read_log(filename)
+    # Define linestyle cycle that will be used for comparison logs.
+    linestyles = ["--","-.",":"]
+
     # Parse plotting arguments.
     plotted_elements = []
     for cmd in plotting_commands:
         # Check that the command is valid, i.e. that all elements exits. If it is the case, add it to the list.
-        same_subplot = (cmd[0] == ':')
```
Comment (duburcqa) [inline python/jiminy_py/src/jiminy_py/log.py:96]: same_subplot should be kept

File: python/jiminy_py/src/jiminy_py/log.py [hunk 3]
```diff
@@ -89,25 +90,27 @@ def plot_log():
...
-
-        # Expand each element if wildcard tokens are present.
+        headers = cmd.split(":")
+        # Expand each element according to regular expression.
         matching_headers = []
         for h in headers:
             matching_headers.append(sorted(fnmatch.filter(log_data.keys(), h)))
-
         # Get minimum size for number of subplots.
-        if same_subplot:
```
Comment (duburcqa) [inline python/jiminy_py/src/jiminy_py/log.py:105]: same

**Comment by matthieuvigne** [comment]
@duburcqa Resolved same-subplot option, sorry about that.

**Pull request merged**